src/pint/toa.py

```diff
@@ -1438,6 +1453,32 @@ def get_flags(self):
         """Return a numpy array of the TOA flags."""
         return self.table["flags"]
 
+    def add_flags(self, indices, flag, flag_values, overwrite=False):
```

Comment:
This replicates functionality in #1074 and adds to the complexity of TOAs objects.